base/array.jl

```diff
@@ -1251,10 +1251,11 @@ julia> findnext(A,3)
 function findnext(A, start::Integer)
     for i = start:length(A)
         if A[i] != 0
-            return i
```

Comment:
I'm not sure those are exactly equivalent, since with this feature you get the default value if the loop never runs.

This also makes it easier to detect the no-break case, which otherwise there is no easy way to do.